src/databases/base.ts

```diff
@@ -0,0 +1,35 @@
+export interface IBaseDatabase {
```

Comment:
Do we have a convention around interface naming? In other codebases I have seen them not being prefixed with `I` and I think I [prefer that style](https://github.com/basarat/typescript-book/blob/master/docs/styleguide/styleguide.md#interface), but I am totally fine with this if we've used this pattern elsewhere.

Reply:
Thanks for the suggestion, I also prefer to not use an `I` but `tslint` was complaining about it, gonna figure out if there's something I can tweak to avoid prepending the `I`

Reply:
Turns out that `tslint` is deprecated, so we should move to use `eslint` instead... 

I'm not going to do the migration just now (too many changes at once), and just disable the `tslint` rule that enforces the `I` prefix on interfaces.